storage/src/mmr/verification.rs

```diff
@@ -234,11 +235,13 @@ impl<H: CHasher> Proof<H> {
                 siblings.sort_by(|a, b| b.0.cmp(&a.0));
             }
         }
-        hashes.extend(
+        let sibling_hashes: Vec<H::Digest> = futures::future::join_all(
             siblings
                 .iter()
-                .map(|(_, pos)| mmr.get_node(*pos).unwrap().clone()),
-        );
+                .map(|(_, pos)| async { mmr.get_node(*pos).await.unwrap().clone() }),
```

Comment:
likewise, shouldn't panic on missing node?

Reply:
changed to propagate storage error, but will still panic on missing node.  Still thinking if I should just return ElementPruned error for this case.

Reply:
changed to return ElementPruned if any element is missing.